playground panics when starting with invalid version

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?

```
[dvaneeden@dve-carbon ~]$ tiup playground --tiflash 0 --without-monitor v9.9.9
```

2. What did you expect to see?

```
Error: Playground bootstrapping failed: version v9.9.9 on linux/amd64 for component pd not found: unknown version
```

3. What did you see instead?

```
tiup is checking updates for component playground ...
Starting component `playground`: /home/dvaneeden/.tiup/components/playground/v1.10.2/tiup-playground --tiflash 0 --without-monitor v9.9.9
Playground Bootstrapping...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xe6434b]

goroutine 1 [running]:
main.(*Playground).terminate(0xc000762000, 0x9)
	github.com/pingcap/tiup/components/playground/playground.go:1000 +0x5eb
main.execute.func3(0xc000740000?, {0xc00072c1c0, 0x1, 0x4?})
	github.com/pingcap/tiup/components/playground/main.go:291 +0x912
github.com/spf13/cobra.(*Command).execute(0xc000740000, {0xc00003a0b0, 0x4, 0x4})
	github.com/spf13/cobra@v1.4.0/command.go:856 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0xc000740000)
	github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.4.0/command.go:902
main.execute()
	github.com/pingcap/tiup/components/playground/main.go:352 +0x1032
main.main()
	github.com/pingcap/tiup/components/playground/main.go:629 +0x3f
```

4. What version of TiUP are you using (`tiup --version`)?


This might be the same issue as #1921

--- a/components/playground/playground.go
+++ b/components/playground/playground.go
@@ -987,26 +987,40 @@ func (p *Playground) terminate(sig syscall.Signal) {
 		go kill("grafana", p.grafana.cmd.Process.Pid, p.grafana.wait)
 	}
 	for _, inst := range p.tiflashs {
-		kill(inst.Component(), inst.Pid(), inst.Wait)
+		if inst.Process != nil {
+			kill(inst.Component(), inst.Pid(), inst.Wait)
+		}
 	}
 	for _, inst := range p.ticdcs {
-		kill(inst.Component(), inst.Pid(), inst.Wait)
+		if inst.Process != nil {
+			kill(inst.Component(), inst.Pid(), inst.Wait)
+		}
 	}
 	for _, inst := range p.drainers {
-		kill(inst.Component(), inst.Pid(), inst.Wait)
+		if inst.Process != nil {
+			kill(inst.Component(), inst.Pid(), inst.Wait)
+		}
 	}
 	// tidb must exit earlier then pd
 	for _, inst := range p.tidbs {
-		kill(inst.Component(), inst.Pid(), inst.Wait)
+		if inst.Process != nil {
+			kill(inst.Component(), inst.Pid(), inst.Wait)
+		}
 	}
 	for _, inst := range p.pumps {
-		kill(inst.Component(), inst.Pid(), inst.Wait)
+		if inst.Process != nil {
+			kill(inst.Component(), inst.Pid(), inst.Wait)
+		}
 	}
 	for _, inst := range p.tikvs {
-		kill(inst.Component(), inst.Pid(), inst.Wait)
+		if inst.Process != nil {
+			kill(inst.Component(), inst.Pid(), inst.Wait)
+		}
 	}
 	for _, inst := range p.pds {
-		kill(inst.Component(), inst.Pid(), inst.Wait)
+		if inst.Process != nil {
+			kill(inst.Component(), inst.Pid(), inst.Wait)
+		}
 	}
 }
 

